src/test/run-pass/shift-near-oflo.rs

```diff
@@ -13,9 +13,6 @@
 // Check that we do *not* overflow on a number of edge cases.
 // (compare with test/run-fail/overflowing-{lsh,rsh}*.rs)
 
-// (Work around constant-evaluation)
-fn id<T>(x: T) -> T { x }
```

Comment:
I don't care as much about LLVM optimizing everything out (my main intent is to test the rust codegen paths) -- but having said that, you are probably right that it would be good to prevent LLVM as well.

Still, would marking the `fn` as '#[inline(never)]` not suffice here?